modules/services/mail/mbsync.nix

```diff
@@ -0,0 +1,76 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.mbsync;
+
+  mbsyncOptions = [ "--all" ] ++ optional (cfg.verbose) "--verbose"
+                  ++ optional (cfg.configFile != "")
+                    "--config ${cfg.configFile}";
+
+in {
+
+  options.services.mbsync = {
+    enable = mkEnableOption "mbsync";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.isync;
+      defaultText = literalExample "pkgs.isync";
+      example = literalExample "pkgs.isync";
+      description = "The package to use for the mbsync binary.";
+    };
+
+    startInterval = mkOption {
+      type = types.nullOr types.int;
+      default = 300;
+      example = literalExample "300";
+      description = "Optional key to run mbsync every N seconds";
+    };
+
+    verbose = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Whether mbsync should produce verbose output.
+      '';
+    };
+
+    configFile = mkOption {
+      type = types.str;
+      default = "";
+      description = ''
+        Optional configuration file to link to use instead of
+        the default file (<filename>~/.mbsyncrc</filename>).
+      '';
+    };
+
+    postExec = mkOption {
```

Comment:
How common is this usecase? Since `launchd.<name>.script` it's possible to do this without an explicit option (alltho that's a bit more implicit).

```nix
{ lib, ... }:
with lib;
{
  services.mbsync.enable = true;
  launchd.user.agents.mbsync.script = mkAfter ''
    mu index
  '';
}
```

Reply:
Personally, I'd say this is a common usecase. Since mbsync will just pull down the e-mail onto the filesystem, most users would want to invoke some sort of mail indexing utility.

I'd vouch for keeping the option :)